keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -413,35 +476,71 @@ class TitusClusterHandler(
       }
     }
 
-  private suspend fun CloudDriverService.getServerGroups(resource: Resource<TitusClusterSpec>): Iterable<TitusServerGroup> =
-    getServerGroups(
+  private suspend fun CloudDriverService.getActiveServerGroups(resource: Resource<TitusClusterSpec>): Iterable<TitusServerGroup> {
+    val existingServerGroups: Map<String, List<ClouddriverTitusServerGroup>> = getExistingServerGroupsByRegion(resource)
+    val activeServerGroups = getActiveServerGroups(
       resource.spec.locations.account,
       resource.spec.moniker,
       resource.spec.locations.regions.map { it.name }.toSet(),
       resource.serviceAccount
-    )
-      .also { them ->
-        val sameContainer: Boolean = them.distinctBy { it.container.digest }.size == 1
-        val healthy: Boolean = them.all {
-          it.instanceCounts?.isHealthy(resource.spec.deployWith.health) == true
+    ).map { activeServerGroup ->
+      // get all server groups in the cluster, split by region,
+      var numEnabled = 0
+      existingServerGroups
+        .getOrDefault(activeServerGroup.location.region, emptyList<ServerGroup>())
+        .forEach { serverGroup ->
+          if (!serverGroup.disabled) {
+            numEnabled += 1
+          }
         }
-        if (sameContainer && healthy) {
-          // only publish a successfully deployed event if the server group is healthy
-          val container = them.first().container
-          getTagsForDigest(container, resource.spec.locations.account)
-            .forEach { tag ->
-              // We publish an event for each tag that matches the digest
-              // so that we handle the tags like `latest` where more than one tags have the same digest
-              // and we don't care about some of them.
-              publisher.publishEvent(
-                ArtifactVersionDeployed(
-                  resourceId = resource.id,
-                  artifactVersion = tag
-                )
-              )
-            }
+      when (numEnabled) {
+        1 -> activeServerGroup.copy(onlyActiveServerGroup = true)
```

Comment:
Shouldn't this be `> 1`?

Reply:
No, if there are more than on enabled server groups that it's not the only enabled one

Reply:
Oof, that's confusing, sorry... Why do we loop and count if all we care about is if there's only one, then? Wouldn't that just be a `.filter` and a `.size`?

Reply:
ha ya that would be better! let me make that change